dataprofiler/tests/data_readers/test_data.py

```diff
@@ -14,6 +15,52 @@ def test_read_data(self):
         for data_type, none_type in zip(data_types, none_types):
             Data(data=none_type, data_type=data_type)
 
+class TestDataReadFromURL(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        
+        http_url = 'https://raw.githubusercontent.com/capitalone/DataProfiler/main/dataprofiler/tests/data/'
+        cls.input_url_names = [
+            dict(path=http_url+'csv/diamonds.csv',
+                 count=53940, delimiter=',', has_header=[0],
+                 num_columns=10, encoding='utf-8', data_type='csv'),
+            dict(path=http_url+'csv/iris-utf-16.csv',
+                 count=150, delimiter=',', has_header=[0],
+                 num_columns=6, encoding='utf-16', data_type='csv'),
+            dict(path=http_url+'avro/users.avro', count=4,  data_type='avro'),
+            dict(path=http_url+'avro/userdata1.avro', count=1000,  data_type='avro'),
+            dict(path=http_url+'json/iris-utf-8.json', encoding='utf-8', count=150, data_type='json'),
+            dict(path=http_url+'json/iris-utf-16.json', encoding='utf-16', count=150, data_type='json'),
+            dict(path=http_url+'parquet/iris.parq', count=150, data_type='parquet'),
+            dict(path=http_url+'parquet/titanic.parq', count=1316, data_type='parquet')
```

Comment:
prolly just limit to:
```
dict(path=test_dir+'/csv/diamonds.csv',
     count=53940, delimiter=',', has_header=[0],
     num_columns=10, encoding='utf-8', data_type='csv'),
dict(path=test_dir+'/avro/users.avro', count=4,  data_type='avro'),
dict(path=test_dir+'/json/iris-utf-16.json', encoding='utf-16', count=150, data_type='json'),
dict(path=test_dir+'/parquet/iris.parq', count=150, data_type='parquet'),
```
that way we test each data_type and a few different encodings.

Reply:
^^ prob can also remove superfluous info as well e.g. num_cols, has_header. however we might be able to check things like count to validate read properly.

we also need to add a `text` data type